cmd/kubectl-gadget/filetop.go

```diff
@@ -0,0 +1,262 @@
+
+	"github.com/kinvolk/inspektor-gadget/cmd/kubectl-gadget/utils"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets/filetop/types"
+)
+
+var (
+	nodeStats map[string][]types.Stats
+	mu        sync.Mutex
+	done      bool = false
+)
+
+var (
+	// arguments
+	interval int = 1
+
+	// flags
+	maxrows    int
+	sortby_str string
+	sortby     types.SortBy
+)
+
+var filetopCmd = &cobra.Command{
+	Use:   "filetop",
+	Short: "Trace reads and writes by file, with container details",
+	Run: func(cmd *cobra.Command, args []string) {
+		var err error
+
+		nodeStats = make(map[string][]types.Stats)
+
+		if len(args) >= 1 {
+			interval, err = strconv.Atoi(args[0])
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "%q is not a valid value for interval\n", args[0])
+				os.Exit(1)
+			}
+		}
+
+		config := &utils.TraceConfig{
+			GadgetName:       "filetop",
+			Operation:        "start",
+			TraceOutputMode:  "Stream",
+			TraceOutputState: "Started",
+			CommonFlags:      &params,
+			Parameters: map[string]string{
+				"output_rows": strconv.Itoa(maxrows),
+				"interval":    strconv.Itoa(interval),
+				"sort_by":     sortby_str,
+			},
+		}
+
+		startprint()
+
+		if err := utils.RunTraceStreamCallback(config, callback); err != nil {
+			fmt.Fprintf(os.Stderr, "%s\n", err)
+			os.Exit(1)
+		}
+	},
+	SilenceUsage: true,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		var err error
+		sortby, err = types.ParseSortBy(sortby_str)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	},
+	Args: cobra.MaximumNArgs(1),
+}
+
+func init() {
+	filetopCmd.Flags().IntVarP(&maxrows, "maxrows", "r", 20, "maximum rows to print")
+	filetopCmd.Flags().StringVarP(&sortby_str, "sort", "", "rbytes", "sort column")
+
+	rootCmd.AddCommand(filetopCmd)
+	utils.AddCommonFlags(filetopCmd, &params)
+}
+
+func callback(line string, node string) {
+	mu.Lock()
+	defer mu.Unlock()
+
+	var event types.Event
+
+	if err := json.Unmarshal([]byte(line), &event); err != nil {
+		fmt.Fprintf(os.Stderr, "error unmarshalling json: %s", err)
+		return
+	}
+
+	if event.Error != "" {
+		fmt.Fprintf(os.Stderr, "error in node %s:%s", event.Node, event.Error)
+		return
+	}
+
+	nodeStats[node] = event.Stats
+}
+
+func startprint() {
+	ticker := time.NewTicker(time.Duration(interval) * time.Second)
+
+	go func() {
+		for {
+			print()
+			_ = <-ticker.C
+		}
+	}()
+}
+
+func print() {
+	// sort and print events
+	mu.Lock()
+
+	stats := []types.Stats{}
+	for _, stat := range nodeStats {
+		stats = append(stats, stat...)
+	}
+	nodeStats = make(map[string][]types.Stats)
+
+	mu.Unlock()
+
+	types.SortStats(stats, sortby)
+
+	switch params.OutputMode {
+	case utils.OutputModeColumns:
+		if term.IsTerminal(int(os.Stdout.Fd())) {
+			utils.ClearScreen()
+		} else {
+			fmt.Println("")
+		}
+		fmt.Printf("%-16s %-16s %-16s %-16s %-7s %-16s %-6s %-6s %-7s %-7s %1s %s\n",
+			"NODE", "NAMESPACE", "POD", "CONTAINER",
+			"TID", "COMM", "READS", "WRITES", "R_Kb", "W_Kb", "T", "FILE")
```

Comment:
From my test with the gadget, I was not able to have a line which shows both `READS > 0 && WRITES > 0`.
So, if this case never occur, would it be possible to replace `R_kb` and `W_kb` by just `kb`?
Maybe I am wrong and I did not play with the gadget enough time, what do you think about it?